compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -515,6 +516,73 @@ SortOp::reifyResultShapes(OpBuilder &b,
       .reifyResultShapes(b, reifiedReturnShapes);
 }
 
+namespace {
+struct RemoveUnusedSortOpResults
+    : public OpRewritePattern<IREE::LinalgExt::SortOp> {
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(IREE::LinalgExt::SortOp sortOp,
+                                PatternRewriter &rewriter) const override {
+    // To avoid problems in dispatches associated with unused results, prune
+    // them here.
+    Location loc = sortOp->getLoc();
+    auto operands = sortOp.getOutputs();
+    auto results = sortOp.getResults();
+    unsigned numRes = sortOp.getNumResults();
+
+    // If there are no results, bufferization has happened.
+    // Do not try to prune results anymore.
+    if (numRes == 0) {
+      return failure();
+    }
```

Comment:
nit: I think what you're looking for is `if (!sortOp.hasPureTensorSemantics())`.

The comment needs to be updated. There are no bufferization context in the op definition.

Question: why can't this happen on buffer semantics? From the op's perspective, it is valid to drop the operands, right?

EDIT: I see what you're doing. Because we need to track the SSA uses. In this case, maybe add a TODO item for it.